cmd/byohctl/service/agent.go

```diff
@@ -0,0 +1,169 @@
+	ByohConfigDir = ".byoh"
+
+	ImgPkgURL  = "https://github.com/carvel-dev/imgpkg/releases/download/v0.45.0/imgpkg-linux-amd64"
+	ImgPkgPath = "/usr/local/bin/imgpkg"
+)
+
+// SetupAgent installs the BYOH agent in the host
+func SetupAgent(byohDirPath string) error {
+	utils.LogInfo("Setting up BYOH agent")
+
+	// Install all pre-requisite packages first
+	utils.LogInfo("Checking and installing required packages...")
+	if err := ensureRequiredPackages(); err != nil {
+		// Since all packages are important, return an error here
+		return fmt.Errorf("failed to install required packages: %v", err)
+	}
+
+	// Proceed with downloading the agent package
+	utils.LogInfo("Downloading agent package...")
+	packagePath, err := downloadDebianPackage(byohDirPath)
+	if err != nil {
+		return fmt.Errorf("failed to download Debian package: %v", err)
+	}
+
+	// Install the agent package
+	utils.LogInfo("Installing BYOH agent package...")
+	if err = installDebianPackage(packagePath); err != nil {
+		return fmt.Errorf("failed to install Debian package: %v", err)
+	}
+
+	utils.LogSuccess("Agent setup completed successfully")
+	return nil
+}
+
+// PrepareAgentDirectory prepares the BYOH agent directory
+func PrepareAgentDirectory(byohDir string) error {
+	// Create byohDir if it doesn't exist
+	if err := os.MkdirAll(byohDir, DefaultDirPerms); err != nil {
+		return fmt.Errorf("failed to create BYOH directory %s: %v", byohDir, err)
+	}
+	return nil
+}
+
+var ensureRequiredPackages = func() error {
+	utils.LogInfo("Checking for required packages...")
+
+	// Fix any broken package state first
+	exec.Command("apt-get", "--fix-broken", "install", "-y").Run()
+
+	// Install imgpkg if needed
+	if _, err := exec.LookPath("imgpkg"); err != nil {
+		utils.LogInfo("Installing imgpkg...")
+
+		// Download imgpkg binar
+
+		// Simple HTTP GET request
+		resp, err := http.Get(ImgPkgURL)
+		if err != nil {
+			return fmt.Errorf("failed to download imgpkg: %v", err)
+		}
+		defer resp.Body.Close()
+
+		// Create the file
+		out, err := os.Create(ImgPkgPath)
+		if err != nil {
+			return fmt.Errorf("failed to create file: %v", err)
+		}
+		defer out.Close()
+
+		// Copy data to file
+		_, err = io.Copy(out, resp.Body)
+		if err != nil {
+			return fmt.Errorf("failed to write file: %v", err)
+		}
+
+		// Make executable
+		os.Chmod(ImgPkgPath, 0755)
+
+		utils.LogSuccess("Installed imgpkg v0.45.0")
+	}
+
+	// Install all required packages in one command
+	utils.LogInfo("Installing required packages...")
+	cmd := exec.Command("bash", "-c",
+		"apt-get update && apt-get install -y --no-install-recommends dpkg ebtables conntrack socat libseccomp2")
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to install packages: %v\nOutput: %s", err, string(output))
+	}
+
+	utils.LogSuccess("All required packages installed successfully")
+	return nil
+}
+
+var downloadDebianPackage = func(tempDir string) (string, error) {
```

Comment:
Why does this need to be assigned to a variable?

Reply:
This is required to mock the installation in unit tests